heat/core/tests/test_types.py

```diff
@@ -0,0 +1,95 @@
+import torch
+import unittest
+
+import heat as ht
+
+
+class TestTypes(unittest.TestCase):
+    def assert_is_heat_type(self, heat_type):
+        self.assertIsInstance(heat_type, type)
+        self.assertTrue(issubclass(heat_type, ht.generic))
+
+    def assert_non_instantiable_heat_type(self, heat_type):
+        self.assert_is_heat_type(heat_type)
+        with self.assertRaises(TypeError):
+            heat_type()
+
+    def assert_is_instantiable_heat_type(self, heat_type, torch_type):
+        # ensure the correct type hierarchy
+        self.assert_is_heat_type(heat_type)
+
+        # check a type constructor without any value
+        no_value = heat_type()
+        self.assertIsInstance(no_value, ht.tensor)
+        self.assertEqual(no_value.shape, (1,))
+        self.assertEqual((no_value._tensor__array == 0).all().item(), 1)
+        self.assertEqual(no_value._tensor__array.dtype, torch_type)
+
+        # check a type constructor with a complex value
```

Comment:
Complex value reminds me too much of a complex number. Wouldn't something like "elaborate input array" fit better, or just `elaborate_array` in order to have a short variable name?